lib/matplotlib/backends/backend_macosx.py

```diff
@@ -112,11 +112,10 @@ def close(self):
 class NavigationToolbar2Mac(_macosx.NavigationToolbar2, NavigationToolbar2):
 
     def __init__(self, canvas):
-        self.canvas = canvas  # Needed by the _macosx __init__.
```

Comment:
Does this need to be set somewhere else, or is passing it in to the constructor all that's really needed?

Reply:
re rect.size: yup.
re self.canvas: `NavigationToolbar2.__init__` takes care of setting the attribute.